Automattic/wp-calypso
# My Home Upsell: use new domain upsell flow

Fixes https://github.com/Automattic/dotcom-forge/issues/1707

## Proposed Changes

Show the new domain upsell flow when clicking on both buttons and show a warning when no plan is selected.

- [x] `Search for a domain` should show the new flow but without the `Not sure yet? Decide later.` text.
- [x] On the plans page (clicking on `Buy this domain` or selecting a domain after 👆), should show a different copy.
- [x] Plans page should show the interval switcher (Monthly / Yearly), and should keep the same flow.
- [x] Clicking on `Or continue with the free plan.` should open a Warning.
- [x] Choosing to not get a plan should show the checkout page with only the domain.
- [x] `< Back` on the plans page should redirect to home coming from `Buy this domain, and to domains coming from the `Search for a domain`

## Testing Instructions

- Use a site with a free plan without custom domain names.
- On the homepage, you should see the Upsell like the Screenshot
- Clocking on `Search for a domain` or `Buy this domain` should show the new domain flow.
- On the plans page, it should show the new copy.
- Clicking on `Or continue with the free plan.` should open a Dialog.
- Clicking on `That works for me` should show the checkout page with only the domain.

### `test-client` testing instructions
- Run `yarn run test-client client/my-sites/customer-home/cards/features/domain-upsell/test/index.jsx` and make sure the test passes
- Ensure this test covers the previous change [this](https://github.com/Automattic/wp-calypso/pull/73326/files) and [this](https://github.com/Automattic/wp-calypso/pull/73391/files) tasks

### `E2E test` testing instructions
- First of all, build the calypso-e2e package with `yarn build` on `/packages/calypso-e2e`.
- Follow [this guide](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/README.md#quick-start) to make sure to have the environment working.
- Run `yarn workspace wp-e2e-tests test -- domain-upsell/domain-upsell__my-home`
- Make sure all test passes

## Screenshots
| Home Upsell | Old Flow | New Flow (domains) | New Flow (plans) | Skip button | Checkout skipping |
| --- | --- | --- | --- | --- | --- |
| ![image](https://user-images.githubusercontent.com/402286/221683228-284f0216-699b-4d78-8c9d-2060e253ab9e.png) | ![image](https://user-images.githubusercontent.com/402286/221683298-f81db090-9fbf-4165-b7db-03cab20231b9.png) | ![image](https://user-images.githubusercontent.com/402286/221683663-80195f64-e881-4614-95b6-368a48bd271a.png) | ![image](https://user-images.githubusercontent.com/402286/222002973-fea79007-37ec-45b4-bc37-b4171ce29f14.png) | ![image](https://user-images.githubusercontent.com/402286/221683358-4699f370-f857-4a59-8a2e-58adf97390a5.png) | ![image](https://user-images.githubusercontent.com/402286/221683375-4c123688-1045-4219-a169-3c675c4ce041.png) |

#### Domains
| Old copy | New Copy |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/402286/222002640-3b39ef22-292a-4e82-b684-d55a128ca958.png) | ![image](https://user-images.githubusercontent.com/402286/222002694-f29c0397-3c6b-4140-94e0-ef71a5cad7a7.png) |



## Pre-merge Checklist
